bayesian/utils.py

```diff
@@ -198,23 +198,22 @@ def generate_recommendation(data, package, version):
                     reco['recommendation'] = {}
                     return {"result": reco}
 
-        # check if latest version exists or current version is latest version
-        if not latest_version or latest_version == '' or version == latest_version:
-            if message != '':
-                reco['recommendation']['message'] = message
-            return {"result": reco}
-        # check if latest version has lower CVEs or no CVEs than current version
         for records in data:
             ver = records['version']
-            if latest_version == ver.get('version', [''])[0]:
-                if ver.get('cve_ids', [''])[0] != '':
-                    for cve in ver.get('cve_ids'):
-                        cvss = float(cve.split(':')[1])
-                        if cvss >= max_cvss:
-                            break
-                message += '\n It is recommended to use Version - ' + latest_version
-                reco['recommendation']['change_to'] = latest_version
-                reco['recommendation']['message'] = message
+            graph_version = ver.get('version', [''])[0]
+            graph_ver = convert_version_to_proper_semantic(graph_version)
+
+            # Check for next best higher version than input version without any CVE's
+            if (ver.get('cve_ids', [''])[0] != '' or 'cve_ids' not in ver) \
```

Comment:
Hopefully this can wait till Monday, because my tired brain is failing to process this.

Looks like `'if (cve_ids is there and not empty) or it's not there'`, which I don't understand.

Reply:
it would help to have such code covered by unit tests ;) is it possible to spend some time on them?

Reply:
You are right. I wanted `if (cve_ids is there and empty) or it's not there`. Fixed it

Reply:
@tisnik - I fixed the if condition but what I was trying with this condition is to make it future change prone if we make our graphDB consistent (create properties will null values). I don't think we require a unit test here.

Reply:
Ok, then I think we can simplify this to `if not ver.get('cve_ids'):` which should do the same, i.e. be true if the `'cve_ids'` is not there or is empty (or None).